### PR TITLE
*: enable error check for some files

### PR DESCRIPTION
**Pull request by okJiang**
### What problem does this PR solve?

Issue Number: Ref https://github.com/tikv/pd/issues/1919

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/schedule/placement/rule.go [hunk 1]
```diff
@@ -90,9 +90,7 @@ func (r *Rule) String() string {
 // Clone returns a copy of Rule.
 func (r *Rule) Clone() *Rule {
 	var clone Rule
-	json.Unmarshal([]byte(r.String()), &clone)
-	clone.StartKey = append(r.StartKey[:0:0], r.StartKey...)
-	clone.EndKey = append(r.EndKey[:0:0], r.EndKey...)
```
Comment (okJiang) [inline pkg/schedule/placement/rule.go:95]: After using `Unmarshal`, there's no need to reallocate the address.

File: pkg/tso/allocator_manager.go [hunk 2]
```diff
@@ -670,44 +663,26 @@ func (am *AllocatorManager) campaignAllocatorLeader(
 	defer am.ResetAllocatorGroup(allocator.GetDCLocation())
 	// Maintain the Local TSO Allocator leader
 	go allocator.KeepAllocatorLeader(ctx)
-	log.Info("campaign local tso allocator leader ok",
-		logutil.CondUint32("keyspace-group-id", am.kgID, am.kgID > 0),
-		zap.String("dc-location", allocator.GetDCLocation()),
-		zap.Any("dc-location-info", dcLocationInfo),
-		zap.String("name", am.member.Name()))
+	logger.Info("campaign local tso allocator leader ok")
```
Comment (rleungx) [inline pkg/tso/allocator_manager.go:666]: How about merging lines 666 and 668, or leaving only one of them?

File: plugin/scheduler_example/evict_leader.go [hunk 3]
```diff
@@ -273,47 +273,55 @@ func (handler *evictLeaderHandler) UpdateConfig(w http.ResponseWriter, r *http.R
...
+	handler.config.mu.Unlock()
+	if err := handler.config.Persist(); err != nil {
 		handler.config.mu.Lock()
-
-		var resp any
-		if len(handler.config.StoreIDWitRanges) == 0 {
-			resp = noStoreInSchedulerInfo
-		}
-		handler.rd.JSON(w, http.StatusOK, resp)
+		_ = handler.rd.JSON(w, http.StatusInternalServerError, err.Error())
```
Comment (rleungx) [inline plugin/scheduler_example/evict_leader.go:315]: Should we rollback the state in the memory?

**Review by rleungx — APPROVED** [review]

File: plugin/scheduler_example/evict_leader.go [hunk 4]
```diff
@@ -273,47 +273,55 @@ func (handler *evictLeaderHandler) UpdateConfig(w http.ResponseWriter, r *http.R
...
 	}
 
-	handler.config.BuildWithArgs(args)
-	err := handler.config.Persist()
+	err := handler.config.BuildWithArgs(args)
 	if err != nil {
-		handler.rd.JSON(w, http.StatusInternalServerError, err.Error())
+		_ = handler.rd.JSON(w, http.StatusBadRequest, err.Error())
+		return
 	}
```
Comment (HuSharp) [inline plugin/scheduler_example/evict_leader.go:280]: How about using `if err := BuildWithArgs(); err != nil` ?

File: pkg/schedule/placement/rule.go (hunk 1 above)
Comment (HuSharp) [reply pkg/schedule/placement/rule.go:95]: startKey is ``json:"-"`` which means cannot be Unmarshal

Comment (okJiang) [reply pkg/schedule/placement/rule.go:95]: Good catch!

File: plugin/scheduler_example/evict_leader.go (hunk 4 above)
Comment (okJiang) [reply plugin/scheduler_example/evict_leader.go:280]: Unless all places maintain a habit, either one is fine, no big deal

Comment (HuSharp) [reply plugin/scheduler_example/evict_leader.go:280]: I just noticed that you are using two methods to check for errors😛 
https://github.com/tikv/pd/pull/8301/files#diff-e5d980e123257fd28befe0f6e57d03f2988cacc60a40a22a76afec251f6e53daR313

**Review by HuSharp — APPROVED** [review]

File: plugin/scheduler_example/evict_leader.go (hunk 4 above)
Comment (okJiang) [reply plugin/scheduler_example/evict_leader.go:280]: I mean all repo keep one habit~

**Pull request merged**